File: vane-proxy-core/src/main/java/org/oddlama/vane/proxycore/ProxyPendingConnection.java
package org.oddlama.vane.proxycore;

import java.util.UUID;
import org.jetbrains.annotations.Nullable;

public interface ProxyPendingConnection {
    String getName();

    @Nullable
    UUID getUniqueId();

    int getPort();

    java.net.SocketAddress getSocketAddress();

    boolean hasPermission(ProxyServer server, final String... permission);

    default boolean canStartServer(ProxyServer server, String serverName) {
        return hasPermission(
            server,
            "vane_proxy.start_server",
            "vane_proxy.start_server.*",
            "vane_proxy.start_server." + serverName
        );
    }
}

File: vane-proxy-core/src/main/kotlin/org/oddlama/vane/proxycore/ProxyPendingConnection.kt
package org.oddlama.vane.proxycore

import java.net.SocketAddress
import java.util.UUID

interface ProxyPendingConnection {
    val name: String?

    val uniqueId: UUID?

    val port: Int

    val socketAddress: SocketAddress?

    fun hasPermission(server: ProxyServer?, vararg permission: String?): Boolean

    fun canStartServer(server: ProxyServer?, serverName: String?): Boolean {
        return hasPermission(
            server,
            "vane_proxy.start_server",
            "vane_proxy.start_server.*",
            "vane_proxy.start_server.$serverName"
        )
    }
}
